File: wordtest.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import sys
reload(sys)
sys.setdefaultencoding('utf-8')
import codecs
from kimmo import *
from wordglossing import *

sargs = sys.argv[:]
wordcorpusfn = 'japanese-words.txt'
if "corpus" in sargs:
	sargs.remove("corpus")
	wordcorpusfn = 'corpus/corpus-allwords-base-pos.txt'
passedword = ""
if len(sargs) > 1:
	passedword = sargs[1].decode("utf-8")

def printWordGloss(k, passedword):
	baseform,pos,otherfeatures = get_base_pos_gloss(k, line)
	if baseform == None or pos == None or otherfeatures == None:
		print passedword + " [FAILURE]"
	else:
		print passedword + " BASE:" + baseform + " POS:" + pos + " " + " ".join(otherfeatures)

k = KimmoRuleSet.load('japanese.yaml')
if passedword != "":
	printWordGloss(k, passedword)
	exit(0)

recfile = codecs.open(wordcorpusfn, encoding='utf-8')
for line in recfile:
	line = line.strip()
	if not line: continue
	if line.startswith(';'):
		print line
		continue
	if " " in line:
		line = line.split(" ")[0]
	printWordGloss(k, line)

recfile.close()

File: wordtest.rb
#!/usr/bin/ruby1.9
# encoding: utf-8

usedictionary = ARGV.include?("usedictionary")
if usedictionary
	ARGV.delete("usedictionary")
end
noregen = ARGV.include?("noregen")
if noregen
	ARGV.delete("noregen")
end
corpus = ARGV.include?("corpus")
if corpus
	ARGV.delete("corpus")
end
passedword = ""
if ARGV.length > 0
	passedword = ARGV[0]
end
if !noregen
	if usedictionary
		system("./generatelexfile.rb usedictionary > japanese-dictionary.lex")
	else
		system("./generatelexfile.rb > japanese-dictionary.lex")
	end
end
system("cat japanese-rules.lex japanese-dictionary.lex japanese-auxiliary.lex > japanese.lex")
#system("./generateyamlfile.rb > japanese.yaml")
if corpus
	system("./wordtest.py corpus")
elsif passedword == ""
	system("./wordtest.py")
else
	system("./wordtest.py #{passedword}")
end

